hazelcast/src/main/java/com/hazelcast/jet/impl/jobupload/JobUploadStatus.java

```diff
@@ -0,0 +1,215 @@
+        this.jobMetaDataParameterObject = parameterObject;
+    }
+
+    public boolean isExpired() {
+        Instant now = Instant.now();
+        Duration between = Duration.between(lastUpdatedTime, now);
+        // Check if total number of minutes is bigger
+        long minutes = between.toMinutes();
+        return minutes >= EXPIRATION_MINUTES;
+    }
+
+    public void onRemove() {
+        try {
+            if (jobMetaDataParameterObject.getJarPath() != null) {
+                Files.delete(jobMetaDataParameterObject.getJarPath());
+            }
+        } catch (IOException exception) {
+            logger.severe("Error while deleting file : " + jobMetaDataParameterObject.getJarPath(), exception);
+        }
+    }
+
+    public JobMetaDataParameterObject processJobMultipart(JobMultiPartParameterObject parameterObject)
+            throws IOException, NoSuchAlgorithmException {
+
+        validateReceivedParameters(parameterObject);
+
+        validateReceivedPartNumbersAreExpected(parameterObject);
+
+        // Parts numbers are good. Save them
+        currentPart = parameterObject.getCurrentPartNumber();
+        totalPart = parameterObject.getTotalPartNumber();
+
+        // If the first part
+        if (currentPart == 1) {
+            createNewTemporaryFile();
+        }
+
+        Path jarPath = jobMetaDataParameterObject.getJarPath();
+
+        // Append data to file
+        try (FileOutputStream outputStream = new FileOutputStream(jarPath.toFile(), true)) {
+            outputStream.write(parameterObject.getPartData(), 0, parameterObject.getPartSize());
+        }
+
+        String message = String.format("Session : %s total file size %d", parameterObject.getSessionId(), Files.size(jarPath));
+        logger.info(message);
+
+        changeLastUpdatedTime();
+
+        JobMetaDataParameterObject result = null;
+        // If parts are complete
+        if (currentPart == totalPart) {
+
+            validateChecksum();
+
+            result = jobMetaDataParameterObject;
+        }
+        return result;
+    }
+
+    private void createNewTemporaryFile() throws IOException {
+        // Create a new temporary file
+        Path jarPath = Files.createTempFile("runjob", ".jar");
+
+        // Make it accessible only by the owner
+        File jarFile = jarPath.toFile();
+        boolean success = jarFile.setReadable(true, true);
+        if (!success) {
+            logger.info("setReadable failed on " + jarFile);
+        }
+        success = jarFile.setWritable(true, true);
+        if (!success) {
+            logger.info("setWritable failed on " + jarFile);
+        }
+        success = jarFile.setExecutable(true, true);
+        if (!success) {
+            logger.info("setExecutable failed on " + jarFile);
+        }
+        jobMetaDataParameterObject.setJarPath(jarPath);
+    }
+
+    private static void validateReceivedParameters(JobMultiPartParameterObject parameterObject) {
```

Comment:
Are the validations below tested anywhere?

Reply:
Added new test